src/process/process.lisp

```diff
@@ -126,36 +126,66 @@ IMPORTANT NOTE: Use #'SPAWN-PROCESS to generate a new PROCESS object."))
-  (setf clauses (append clauses `((message-RTS 'handle-message-RTS))))
-  (a:with-gensyms (node message now)
-    `(defmethod message-dispatch ((,node ,node-type) ,now)
-       ,@(loop :for (message-type receiver . rest) :in clauses
-               :for guard := (or (first rest) t)
-               :collect `(when (let ((,node-type ,node))
-                                 (declare (ignorable ,node-type))
-                                 ,guard)
-                           (receive-message ((process-key ,node) ,message
-                                             :catch-RTS? nil
-                                             :peruse-inbox? (process-peruse-inbox?
-                                                             ,node))
-                             (,message-type
-                              (when (process-debug? ,node)
-                                (log-entry :source-type ',node-type
-                                           :time ,now
-                                           :entry-type 'handler-invoked
-                                           :source (process-public-address ,node)
-                                           :message-id (message-message-id ,message)
-                                           :payload-type ',message-type))
-                              (return-from message-dispatch
-                                (values
-                                 (funcall ,receiver ,node ,message ,now)
-                                 t)))))))))
+  (a:with-gensyms (node message now results trapped?)
+    `(defmethod %message-dispatch ((,node ,node-type) ,now)
+       ,@(mapcar
+          (lambda (clause)
+            (cond
+              ((listp clause)
+               (destructuring-bind (message-type receiver . rest) clause
+                 `(when (let ((,node-type ,node))
+                          (declare (ignorable ,node-type))
+                          ,(or (first rest) t))
+                    (receive-message ((process-key ,node) ,message
+                                      :catch-RTS? nil
+                                      :peruse-inbox? (process-peruse-inbox?
+                                                      ,node))
+                      (,message-type
+                       (when (process-debug? ,node)
+                         (log-entry :source-type ',node-type
+                                    :time ,now
+                                    :entry-type 'handler-invoked
+                                    :source (process-public-address ,node)
+                                    :message-id (message-message-id ,message)
+                                    :payload-type ',message-type))
+                       (return-from %message-dispatch
+                         (values
+                          (funcall ,receiver ,node ,message ,now)
+                          t)))))))
+              ((and (symbolp clause) (string= "CALL-NEXT-METHOD" (symbol-name clause)))
```

Comment:
I'm trying to avoid package capture of the symbol call-next-method, but I'm not sure this is the right way to do it.